tracing/http_test.go

```diff
@@ -69,22 +68,21 @@ func TestHTTPSpanMiddleware(t *testing.T) {
 	next := testHandler{}
 	middleware := HTTPSpanMiddleware(&next)
 
-	var lastWrite []byte
-	fw := fakeWriter{lastWrite: &lastWrite}
+	fw := &fakeWriter{}
 
 	req, err := http.NewRequest("GET", "http://test.example.com", nil)
 	if err != nil {
-		t.Errorf("Failed to make fake request: %v", err)
+		t.Fatal("Failed to make fake request:", err)
 	}
-	traceID := "821e0d50d931235a5ba3fa42eddddd8f"
+	const traceID = "821e0d50d931235a5ba3fa42eddddd8f"
 	req.Header["X-B3-Traceid"] = []string{traceID}
 	req.Header["X-B3-Spanid"] = []string{"b3bd5e1c4318c78a"}
 
 	middleware.ServeHTTP(fw, req)
 
 	// Assert our next handler was called
-	if diff := cmp.Diff([]byte("fake"), lastWrite); diff != "" {
-		t.Errorf("Got http response (-want, +got) = %v", diff)
+	if got, want := fw.lastWrite, "fake"; got != want {
```

Comment:
why this over diff? is it a short value?

Reply:
1. short and string, string diffs are not very useful, especially for 4 chars
2. != is cheaper